Src/Support/Google.Apis/Requests/BatchRequest.cs

```diff
@@ -179,7 +179,35 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             var result = await httpClient.PostAsync(new Uri(batchUrl), outerContent, cancellationToken)
                 .ConfigureAwait(false);
 
-            result.EnsureSuccessStatusCode();
+            if (!result.IsSuccessStatusCode)
+            {
+                try
+                {
+                    // Parse the error from the current response.
+                    var error = await service.DeserializeError(result).ConfigureAwait(false);
+                    if (error != null)
+                    {
+                        throw new GoogleApiException(service.Name, error.ToString())
+                        {
+                            Error = error,
+                            HttpStatusCode = result.StatusCode
+                        };
+                    }
+                }
+                catch(Exception apiException)
```

Comment:
More commentary around the circumstances of this would be useful.

Reply:
Yes, and also agree with extracting to another method.